src/App.js

```diff
@@ -23,21 +24,59 @@ const generateSquares = () => {
   }
 
   return squares;
+  
 }
 
 const App = () => {
 
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [winner, setWinner] = useState(null);
 
   // Wave 2
   // You will need to create a method to change the square 
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
+   const onClickCallback=(returningId)=>{
```

Comment:
In this function, you could create a check for how many squares are filled. You can create a state that could be squares filled and setSquaresFilled with an initial state of 0.